src/ucp/proto/proto_perf.h

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_PROTO_PERF_H_
+#define UCP_PROTO_PERF_H_
+
+#include "proto.h"
+
+#include <ucs/datastruct/linear_func.h>
+#include <ucs/datastruct/string_buffer.h>
+#include <ucs/type/status.h>
+
+
+/* Protocol performance data streucture over multiple ranges */
+typedef struct ucp_proto_perf ucp_proto_perf_t;
+
+
+/* Protocol performance segment, defines the performance in a single range */
+typedef struct ucp_proto_perf_segment ucp_proto_perf_segment_t;
+
+
+/* Protocol performance factor type */
+typedef enum {
+    UCP_PROTO_PERF_FACTOR_LOCAL_CPU,
+    UCP_PROTO_PERF_FACTOR_REMOTE_CPU,
+    UCP_PROTO_PERF_FACTOR_TL_SEND,
+    UCP_PROTO_PERF_FACTOR_TL_RECEIVE,
+    UCP_PROTO_PERF_FACTOR_LATENCY,
+    UCP_PROTO_PERF_FACTOR_SINGLE, /* For compatibility; to remove */
+    UCP_PROTO_PERF_FACTOR_MULTI,  /* For compatibility; to remove */
+    UCP_PROTO_PERF_FACTOR_LAST
+} ucp_proto_perf_factor_id_t;
```

Comment:
Regarding "pre" and "post", we can trait CPU factors as sum of "pre" and "post" for MULTI perf case and build envelope basing on this sum. For SINGLE perf case "post" part happens in parallel with all other factors except CPU factor. Both these cases can be handled during SINGLE and/or MULTI perf final envelope calculation if they are separated.

So if we leave that as is, should we include `send_post_overhead` to `LOCAL_CPU` factor?

I am OK to leave it without separation, but as I know cuda copy has `send_post_overhead` equal to 1.8 usec which seems significant overhead, so it can cause performance degradation in some cases.